src/manifolds/MetricManifold.jl

```diff
@@ -51,6 +51,126 @@ inner product ``g(X, X) > 0`` whenever ``X`` is not the zero vector.
+representations ``G_1`` and ``G_2`` and ``x,y`` are the local coordinates with respect to
+the same basis of the tangent space, the equation reads
+
+```math
+   x^*G_2y = c(x)^*G_1 y \quad \text{for all } y \in ℝ^d,
+```
+where `\cdot^*`` denotes the conjugate transpose.
+
+and we obtain ``c(X) = (G_1\backslash G_2)^*X``
+
+# Examples
+
+    change_gradient(Sphere(2), EuclideanMetric(), p, X)
+
+Since the metric in ``T_p\mathbb S^2`` is the Euclidean metric from the embedding restricted to ``T_p\mathbb S^2``, this just returns `X`
+
+    change_gradient(SymmetricPOsitiveDefinite(3), EuclideanMetric(), p, X)
+
+Here, the default metric in `\mathcal P(3)` is the [`LinearAffineMetric`](@ref) and the transformation can be computed as ``pXp``
+"""
+change_gradient(::AbstractManifold, ::AbstractMetric, ::Any, ::Any)
+
+function change_tangent(M::AbstractManifold, G::AbstractMetric, p, X)
+    if is_default_metric(M, G)
+        return X
+    end
+    # TODO: For local metric, inverse_local metric, det_local_metric: Introduce a default basis?
+    B = DefaultOrthogonalBasis()
+    G1 = local_metric(M, p, B)
+    G2 = localMetric(G(M), p, B)
+    x = get_coordinates(M, p, X, B)
+    z = (G2 \ G1)'x
+    return get_vector(M, p, z, B)
+end
+
+function change_metric(
+    ::MetricManifold{𝔽,M,G},
+    ::G,
+    p,
+    X,
+) where {𝔽,M<:AbstractManifold{𝔽},G<:AbstractMetric}
+    return X
+end
+
+@doc raw"""
+    change_metric(M::AbstractcManifold, G2::AbstractMetric, p, X)
+
+On the [`AbstractManifold`](@ref) `M` with implicitly given metric ``g_1``
+and a second [`AbstractMetric`](@ref) ``g_2`` this method performs a change of metric in the
+sense that it returns the tangent vector ``Z=BX`` such that the linear map ``B`` fulfills
+
+````math
+g_2(Y_1,Y_2) = g_1(BY_1,BY_2) \quad \text{for all } Y_1, Y_2 ∈ T_p\mathcal M.
+````
+
+holds.
+
+If both metrics are given in their [`local_metric`](@ref) (symmetric positive defintie) matrix
+representations ``G_1 = C_1C_1^*`` and ``G_2 = C_2C_2^*``, where ``C_1,C_2`` denote their
+Cholesky factor, then solving ``C_2C_2^* = G_2 = B^*G_1B = B^*C_1C_1^*B`` yields ``B = (C_1 \backslash C_2)^*``,
+where `\cdot^*`` denotes the conjugate transpose.
+
+This function returns `Z = BX`.
+
+# Examples
+
+    change_metric(Sphere(2), EuclideanMetric(), p, X)
+
+Since the metric in ``T_p\mathbb S^2`` is the Euclidean metric from the embedding restricted to ``T_p\mathbb S^2``, this just returns `X`
+
+    change_metric(SymmetricPOsitiveDefinite(3), EuclideanMetric, p, X)
+
+Here, the default metric in `\mathcal P(3)` is the [`LinearAffineMetric`](@ref) and the transformation can be computed as ``B=p``
+
+"""
+change_metric(::AbstractManifold, ::AbstractMetric, ::Any, ::Any)
+
+function change_metric(M::AbstractManifold, G::AbstractMetric, p, X)
+    if is_default_metric(M, G)
+        return X
+    end
+    # TODO: For local metric, inverse_local metric, det_local_metric: Introduce a default basis?
```

Comment:
Default bases is something that may be worth updating but it's a bigger issue. Anyway, here we want the same basis for two different metrics so it has to be more or less the way you wrote it.

Reply:
Yes, sure, I would like to have B here as an optional parameter, as well as for the local metic (inverse local metric, yet local metric, Jacobean local metric ...)

Reply:
Yeah, but I wouldn't like to work on that any time soon...

Reply:
me neither.Maybe before merging this we open an issue for this to make a few of these metric functions more flexible / provide a default basis for them.

Reply:
Sure, we can definitely have an issue for that.

Reply:
See #424.